packages/explain-plan-helper/src/getExecutionStats.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  isAggregationExplain,
+  isShardedAggregationExplain,
+  getStageCursorKey,
+} from 'mongodb-explain-compat';
+
+import type { Stage } from './index';
+
+export type ExecutionStats = {
+  executionSuccess: boolean;
+  nReturned: number;
+  executionTimeMillis: number;
+  totalKeysExamined: number;
+  totalDocsExamined: number;
+  executionStages: Stage;
+  allPlansExecution: unknown[];
+}
+
+export const getExecutionStats = (explain: Stage): ExecutionStats => {
+  const executionStats = isAggregationExplain(explain)
+    ? _getAggregationStats(explain)
+    : _getFindStats(explain);
+  return executionStats;
+}
+const _getAggregationStats = (explain: Stage): ExecutionStats => {
+  return isShardedAggregationExplain(explain)
+    ? _getShardedAggregationStats(explain)
+    : _getUnshardedAggregationStats(explain);
+};
+const _getUnshardedAggregationStats = (explain: Stage): ExecutionStats => {
+  const firstStage = explain.stages[0];
+  const cursorKey = getStageCursorKey(firstStage);
+  if (!cursorKey) {
+    throw new Error('Can not find a cursor stage.');
+  }
+
+  const lastStage = explain.stages[explain.stages.length - 1];
+
+  const stats = _getFindStats(firstStage[cursorKey]);
+  stats.nReturned = lastStage.nReturned;
+  stats.executionTimeMillis = sumArrayProp(explain.stages, 'executionTimeMillisEstimate');
+  return stats;
+};
+const _getShardedAggregationStats = (explain: Stage): ExecutionStats => {
+  const shardStats = [];
+  for (const shardName in explain.shards) {
+    const stats = explain.shards[shardName].stages
+      ? _getUnshardedAggregationStats(explain.shards[shardName])
+      : _getFindStats(explain.shards[shardName]);
+
+    shardStats.push({
+      shardName,
+      ...stats
+    });
+  }
+
+  const nReturned = sumArrayProp(shardStats, 'nReturned');
+  const executionTimeMillis = sumArrayProp(shardStats, 'executionTimeMillis');
+  const totalKeysExamined = sumArrayProp(shardStats, 'totalKeysExamined');
+  const totalDocsExamined = sumArrayProp(shardStats, 'totalDocsExamined');
+  const response = {
+    nReturned,
+    executionTimeMillis,
+    totalKeysExamined,
+    totalDocsExamined,
+    allPlansExecution: [],
+    executionSuccess: true,
+    executionStages: {
+      stage: shardStats.length === 1 ? "SINGLE_SHARD" : "SHARD_MERGE",
```

Comment:
Does the linter not complain about this?
```suggestion
      stage: shardStats.length === 1 ? 'SINGLE_SHARD' : 'SHARD_MERGE',
```

Where is this stage name consumed/why these specific values? I can’t see anything outside of tests that references them

Reply:
Great catch! Here i am creating a response similar to shardedFind by merging execution stats from various shards (something similar to [sharded_geo_query_3.2.json](https://github.com/mongodb-js/compass/blob/COMPASS-5786-explain-plan-helper/packages/explain-plan-helper/test/fixtures/sharded_geo_query_3.2.json)). In case we want to recreate a UI similar to Query Explain, we will have all the data already processed. 

Also we need these aggregated stats in Query Performance Summary of Aggregation Explain.